### PR TITLE
feat(dqlite): Add WatchModelMachines on machine WatchableService

### PR DESCRIPTION
**Pull request by cderici**
This adds the following:

- Adds a `WatchableService` on top of the machine service, and implements the `WatchModelMachines` (old one is at `state/watcher.go`) on top of that `WatchableService`.
- Adds triggers for machine table (that puts `uuid` into the change).
- Return `uuid` from `CreateMachine` from the machine state layer
- Lifts up the `uuid` creation from the state layer up to the service layer (if we do retries we don't keep regenerating stuff).
- Renames `WatchModelMachines -> WatchMachines` and `UpsertMachine -> CreateMachine`

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```
 $ TEST_PACKAGES="./domain/schema/... -count=1 -race" make run-go-tests
 $ TEST_PACKAGES="./domain/machine/... -count=1 -race" make run-go-tests
```

## Links

**Jira card:** JUJU-6197

### CODE REVIEW

File: domain/machine/service/watcher.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		Service: Service{
+			st: st,
+		},
+		watcherFactory: watcherFactory,
+	}
+}
+
+// WatchModelMachines returns a StringsWatcher that is subscribed to the changes
+// in the machines table in the model.
+func (s *WatchableService) WatchModelMachines(ctx context.Context) (watcher.StringsWatcher, error) {
```
Comment (tlm) [inline domain/machine/service/watcher.go:40]: Do we think that this name is ok? I think Model is redundant and wouldn't mind seeing if anyone else thinks that word should be removed?

Comment (SimonRichardson) [reply domain/machine/service/watcher.go:40]: Yeah, I agree with @tlm, you can drop `Model`

File: domain/machine/service/service.go [hunk 2]
```diff
@@ -7,12 +7,14 @@ import (
...
 	"github.com/juju/errors"
+
+	"github.com/juju/juju/internal/uuid"
 )
 
 // State describes retrieval and persistence methods for machines.
 type State interface {
 	// UpsertMachine persists the input machine entity.
-	UpsertMachine(context.Context, string) error
+	UpsertMachine(context.Context, string, string, string) error
```
Comment (SimonRichardson) [inline domain/machine/service/service.go:17]: Can you rename this to `CreateMachine`. We know that we're creating a machine, no need to pretend that we might be creating vs updating it.

File: domain/machine/state/state.go [hunk 3]
```diff
@@ -33,7 +32,7 @@ func NewState(factory coredb.TxnRunnerFactory, logger logger.Logger) *State {
 
 // UpsertMachine creates or updates the specified machine.
 // TODO - this just creates a minimal row for now.
-func (st *State) UpsertMachine(ctx context.Context, machineId string) error {
+func (st *State) UpsertMachine(ctx context.Context, machineId, nodeUUID, machineUUID string) error {
```
Comment (SimonRichardson) [inline domain/machine/state/state.go:35]: Rename to `CreateMachine`

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**